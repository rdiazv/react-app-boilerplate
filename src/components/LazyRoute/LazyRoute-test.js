import { React, mount } from 'helpers/test-helper'
import LazyRoute from 'components/LazyRoute'

describe('LazyRoute', () => {
  it('should render', () => {
    const component = mount(
      <LazyRoute
        getComponent={() => new Promise(() => {})}
      />
    )

    expect(component).toMatchSnapshot()
  })
})
