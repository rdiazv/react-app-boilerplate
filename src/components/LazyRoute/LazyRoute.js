import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Route } from 'react-router-dom'
import Lazy from 'components/Lazy'

export default class LazyRoute extends PureComponent {
  static propTypes = {
    getComponent: PropTypes.func.isRequired,
  }

  render() {
    const {Â getComponent, ...other } = this.props

    return (
      <Route
        {...other}
        render={props =>
          <Lazy
            {...props}
            getComponent={getComponent}
          />
        }
      />
    )
  }
}
