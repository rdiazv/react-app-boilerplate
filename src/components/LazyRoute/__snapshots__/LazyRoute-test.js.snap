// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyRoute should render 1`] = `
<LazyRoute
  getComponent={[Function]}
>
  <Route
    render={[Function]}
  >
    <Lazy
      getComponent={[Function]}
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "oucuam",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "oucuam",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      loadingComponent={null}
      location={
        Object {
          "hash": "",
          "key": "oucuam",
          "pathname": "/",
          "search": "",
          "state": undefined,
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
    />
  </Route>
</LazyRoute>
`;
